File: src/ortools/OrTools.Console/Models/Payload/CategoryWindow.cs
using OrTools.Console.Models.Enums;

namespace OrTools.Console.Models.Payload;

public record CategoryWindow(Category Category, DateTime StartDateTime, DateTime EndDateTime)
{
    public Optimizer.Models.Payload.CategoryWindow ToProviderModel() =>
        new(Category.ToProviderModel(), StartDateTime, EndDateTime);
}

File: src/timefold/src/main/java/com/pso/timefold/domain/CategoryWindow.java
package com.pso.timefold.domain;

import com.pso.timefold.domain.enums.Category;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
public class CategoryWindow {

    private Category category;
    private LocalDateTime startDateTime;
    private LocalDateTime endDateTime;

    private int startMinuteFromHorizon = -1;
    private int endMinuteFromHorizon   = -1;
}
